benches/benches/bevy_picking/ray_mesh_intersection.rs

```diff
@@ -14,23 +14,32 @@ struct SimpleMesh {
 }
 
 fn mesh_creation(vertices_per_side: u32) -> SimpleMesh {
-    let mut positions = Vec::new();
-    let mut normals = Vec::new();
-    for p in 0..vertices_per_side.pow(2) {
-        let xz = ptoxznorm(p, vertices_per_side);
-        positions.push([xz.0 - 0.5, 0.0, xz.1 - 0.5]);
-        normals.push([0.0, 1.0, 0.0]);
-    }
+    let vertices_squared = vertices_per_side.pow(2);
+    let positions = (0..vertices_squared)
+        .map(|p| {
+            let xz = ptoxznorm(p, vertices_per_side);
+            [xz.0 - 0.5, 0., xz.1 - 0.5]
+        })
+        .collect();
 
-    let mut indices = vec![];
-    for p in 0..vertices_per_side.pow(2) {
-        if p % (vertices_per_side) != vertices_per_side - 1
-            && p / (vertices_per_side) != vertices_per_side - 1
-        {
-            indices.extend_from_slice(&[p, p + 1, p + vertices_per_side]);
-            indices.extend_from_slice(&[p + vertices_per_side, p + 1, p + vertices_per_side + 1]);
-        }
-    }
+    let normals = std::iter::repeat([0., 1., 0.])
+        .take(vertices_squared as usize)
+        .collect();
+
+    let indices = (0..vertices_squared)
+        .filter(|&p| {
+            p % vertices_per_side != vertices_per_side - 1
+                && p / vertices_per_side != vertices_per_side - 1
+        })
```

Comment:
I believe this filter expression removes `vertices_per_side`-number of items for `p % vertices_per_side != vertices_per_side - 1`, and another `vertices_per_side` worth in `p / vertices_per_side != vertices_per_side - 1`. Meaning the total number of items this produces (after the `flatten`) is between `6 * (vertices_squared - 2 * vertices_per_side)` and `6 * (vertices_squared - vertices_per_side)`. It _might_ be faster to allocate the `Vec` using `Vec::with_capacity(6 * (vertices_squared - vertices_per_side))` (over-estimating to avoid reallocation), since I don't believe a `collect` statement is faster when the size of the iterator isn't knowable ahead of time (which `filter` and `flat_map` can do).

Could also spend a half-hour in Desmos working out the _exact_ length of the `Vec` as a function of `vertices_per_side` to minimise memory usage, but even the overestimate _should_ perform better.